corporate/lib/stripe.py

```diff
@@ -1411,6 +1412,28 @@ def set_required_plan_tier(self, required_plan_tier: int) -> str:
             plan_tier_name = CustomerPlan.name_from_tier(new_plan_tier)
         return f"Required plan tier for {self.billing_entity_display_name} set to {plan_tier_name}."
 
+    def configure_temporary_courtesy_plan(self, end_date_string: str) -> str:
+        plan_end_date = datetime.strptime(end_date_string, "%Y-%m-%d").replace(tzinfo=timezone.utc)
+        if plan_end_date.date() <= timezone_now().date():
+            raise SupportRequestError(
+                f"Cannot configure a courtesy plan for {self.billing_entity_display_name} to end on {end_date_string}."
+            )
+        customer = self.get_customer()
+        if customer is not None:
+            plan = get_current_plan_by_customer(customer)
+            if plan is not None:
+                raise SupportRequestError(
+                    f"Cannot configure a courtesy plan for {self.billing_entity_display_name} because of current plan."
+                )
+        plan_anchor_date = timezone_now()
+        if isinstance(self, RealmBillingSession):
+            raise SupportRequestError(
+                f"Cannot currently configure a courtesy plan for {self.billing_entity_display_name}."
+            )  # nocoverage
+        else:
```

Comment:
Usually one doesn't write an `else` statement if the previous block exited via exception or returning.